src/reader.cpp

```diff
@@ -86,6 +86,15 @@ Reader::Reader(const string zimFilePath)
   srand(time(nullptr));
 }
 
+Reader::Reader(int fd)
+  :  zimArchive(new zim::Archive(fd)),
+     zimFilePath("")
+{
+
+  /* initialize random seed: */
+  srand(time(nullptr));
```

Comment:
I see some inconsistency here. #446 doesn't drop the call to `srand()` in the other (pre-existing) constructor of `kiwix::Reader`. If that was an omission, then #446 can be amended for all three constructors of `kiwix::Reader`. Otherwise, if one constructor of `kiwix::Reader` contains that code, the others should do the same.

Reply:
Yes, see my global comment in the PR. We should remove it in #446.